src/vm/opcodes.rs

```diff
@@ -20,37 +20,288 @@
 // See the License for the specific language governing permissions and
 // limitations under the License.
 
-#![allow(clippy::unusual_byte_groupings)]
+use std::marker::PhantomData;
 
-use aluvm::isa::opcodes::INSTR_ISAE_TO;
+use aluvm::reg::{Reg16, Reg32, RegS};
+use amplify::num::{u2, u3};
 
-pub const INSTR_RGBISA_FROM: u8 = 0b10_010_000;
-pub const INSTR_RGBISA_TO: u8 = INSTR_ISAE_TO;
+use super::ContractStateAccess;
 
-// CONTRACTS:
-pub const INSTR_CNP: u8 = 0b11_000_000;
-pub const INSTR_CNS: u8 = 0b11_000_001;
-pub const INSTR_CNG: u8 = 0b11_000_010;
-pub const INSTR_CNC: u8 = 0b11_000_011;
+/// Operations defined under RGB ISA extension (`RGB`).
+#[derive(Copy, Clone, Ord, PartialOrd, Eq, PartialEq, Hash, Debug, Display)]
+pub enum ContractOp<S: ContractStateAccess> {
+    /// Counts number of global items elements defined by the current operation of the type,
```

Comment:
Please correct me if I'm wrong.

Does this opcode get data from the global state of the contract (coming from the first operation, also known as genesis)?

I suggest changing the comment to clarify the difference between the operations performed by the .c suffix and the .g suffix.


```suggestion
    /// Counts number of global items elements defined by the contract global state of the type,
```

Reply:
You are right, I have committed your suggestion! You have a very good eye!